### ISSUE TITLE
[feat]: add native support for deno 2 as a package manager.

### ISSUE THREAD

**Issue by mattgi** [shadcn-ui/ui#5529]
### Feature description

Deno 2 can be used as a drop in replacement for node and all package managers so please add support for Deno 2 in the CLI.

Code within `init` and `add` commands test only for pnpm, yarn, npm, and bun so will not currently support deno. There are other issues as well, such as the dependency upon package.json and tsconfig.

React projects can run fine with these constraints so people will slowly start to prefer deno over node, given the increased speed and native support for TypeScript.

see https://docs.deno.com/deploy/tutorials/vite/#step-1%3A-create-a-vite-app for a working example of running React without a package.json or tsconfig file.

### Affected component/components

CLI

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by mattgi** [comment]
current output from running `init`:

`deno run npm:shadcn@latest init`
✅ Granted all env access.
✅ Granted all read access.
✅ Granted all sys access.
✔ The path /git/deno/web does not contain a package.json file. Would you like to start a new Next.js project? … no

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to two years of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you.

**Comment by shadcn** [comment]
No stale. Something went wrong with the stale bot.

**Comment by EpochBoy** [comment]
Yeah same issue, I'm trying to get shadcn to work with Deno 2, no luck thus far.  Would be nice to have an easy way of implementing shadcn in a project that uses deno.

`deno run npm:shadcn@latest init
Warning The following packages contained npm lifecycle scripts (preinstall/install/postinstall) that were not executed:
┠─ npm:msw@2.7.3
┃
┠─ This may cause the packages to not work correctly.
┖─ To run lifecycle scripts, use the `--allow-scripts` flag with `deno install`:
   deno install --allow-scripts=npm:msw@2.7.3
✅ Granted env access to "GRACEFUL_FS_PLATFORM".
✅ Granted env access to "TEST_GRACEFUL_FS_GLOBAL_PATCH".
✅ Granted sys access to "cpus".
✅ Granted env access to "FORCE_COLOR".
✅ Granted env access to "NODE_DISABLE_COLORS".
✅ Granted env access to "NO_COLOR".
✅ Granted env access to "TERM".
✅ Granted env access to "TF_BUILD".
✅ Granted env access to "CI".
✅ Granted env access to "TEAMCITY_VERSION".
✅ Granted env access to "COLORTERM".
✅ Granted env access to "READABLE_STREAM".
✅ Granted env access to "TS_ETW_MODULE_PATH".
✅ Granted read access to "SOME_PATH/NODE_MODULES/.DENO/@TS-MORPH+COMMON@0.19.0/NODE_MODULES/@TS-MORPH/COMMON/DIST/TYPESCRIPT.JS".
✅ Granted env access to "TSC_WATCHFILE".
✅ Granted all env access.
✅ Granted all read access.
✅ Granted all sys access.
✖ The path SOME_PATH does not contain a package.json file.
  Would you like to start a new project? › Next.js

Something went wrong. Please check the error below for more details.
If the problem persists, please open an issue on GitHub.

The "path" argument must be of type string. Received undefined`

Would be nice to be able to say no to find the final part about starting a new project.

**Comment by ErikDombi** [comment]
+1 here echoing frustrations with lack of deno 2 support

**Comment by opticSquid** [comment]
Just to not let issue go stale. Shadcn does need deno support

**Comment by shadcn** [comment]
Would the following PR fix it? https://github.com/shadcn-ui/ui/pull/7006

**Comment by opticSquid** [comment]
> Would the following PR fix it? https://github.com/shadcn-ui/ui/pull/7006

Seems like it should. But I would also want others opinion on this

**Comment by idest** [comment]
Support for deno is still needed, and it seems like the working implementation from the mentioned PR is just waiting to be approved.

**Comment by gytis-ivaskevicius** [comment]
It also requires package.json (which should be easy to fix) and tsconfig.json
I've added small dummy files. I hope this is a temporary workaround

**Comment by Sogeman** [comment]
Is this still coming?
